datafusion/optimizer/src/optimize_projections/mod.rs

```diff
@@ -767,7 +788,12 @@ fn rewrite_projection_given_requirements(
     // projection down
     optimize_projections(Arc::unwrap_or_clone(input), config, required_indices)?
         .transform_data(|input| {
-            if is_projection_unnecessary(&input, &exprs_used)? {
+            if !config
+                .options()
+                .optimizer
+                .optimize_projections_preserve_existing_projections
+                && is_projection_unnecessary(&input, &exprs_used)?
```

Comment:
Similarly, pass the config into `is_projection_unnecessary`

Reply:
Updated - thank you